harmoni-web-api/src/main/java/org/harmoniapp/harmoniwebapi/services/MessageService.java

```diff
@@ -0,0 +1,88 @@
+package org.harmoniapp.harmoniwebapi.services;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.harmoniapp.harmonidata.entities.Message;
+import org.harmoniapp.harmonidata.entities.User;
+import org.harmoniapp.harmonidata.repositories.RepositoryCollector;
+import org.harmoniapp.harmoniwebapi.contracts.MessageDto;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@ComponentScan(basePackages = {"org.harmoniapp.harmonidata"})
+public class MessageService {
+    private final RepositoryCollector repositoryCollector;
+    private final SimpMessagingTemplate messagingTemplate;
+    private final TranslationService translationService;
+
+//    public List<MessageDto> getChatHistory(Long userId1, Long userId2) {
+//        List<Message> messages= repositoryCollector.getMessages().findChatHistory(userId1, userId2);
+//        return messages.stream()
+//                .map(MessageDto::fromEntity)
+//                .toList();
+//    }
+
+    public List<MessageDto> getChatHistory(Long userId1, Long userId2, boolean translate, String targetLanguage) {
+        List<Message> messages = repositoryCollector.getMessages().findChatHistory(userId1, userId2);
+
+        return messages.stream()
+                .map(message -> {
+                    String content = message.getContent();
+
+                    if (translate && targetLanguage != null && !targetLanguage.isEmpty()) {
+                        content = translationService.translate(message.getContent(), targetLanguage);
+                    }
+
+                    return new MessageDto(
+                            message.getId(),
+                            message.getSender().getId(),
+                            message.getReceiver().getId(),
+                            content,
+                            message.getSentAt(),
+                            message.isRead()
+                    );
+                }).toList();
+    }
```

Comment:
Chat should be with pagination or other division such as date range. Currently, it is loaded in its entirety, which will eventually heavily increase the processing time of the query in case of a large number of messages.